File: arrays/search/validMountainArray.java
package search;

public class validMountainArray {
  public boolean validation(int[] arr) {
    if (arr.length < 3) {
      return false;
    }
    int biggestIntIndex = 0;
    for (int i = 0; i < arr.length; i++) {
      if (arr[i] > biggestIntIndex) {
        biggestIntIndex = i;
      }
    }
    for (int j = 0; j < biggestIntIndex - 1; j++) {
      if (arr[j + 1] - arr[j] <= 0) {
        return false;
      }
    }
    for (int k = biggestIntIndex; k < arr.length - 1; k++) {
      if (arr[k] - arr[k + 1] <= 0) {
        return false;
      }
    }
    return true;
  }
}

File: arrays/search/validMountainArray.cpp
#include <iostream>
#include <vector>

class Solution {
public:
  bool validMountainArray(std::vector<int> &arr) {
    if (arr.size() < 3)
      return false;

    int i = 0;
    int N = arr.size();

    while (i + 1 < N - 1 && arr[i] < arr[i + 1])
      i++;

    if (i == 0 || i == N - 1)
      return false;

    while (i + 1 < N - 1 && arr[i] > arr[i + 1])
      i++;
    return i == N - 1;
  }
};

int main() {
  Solution sol;
  std::vector<int> arr = {3, 5, 4};
  std::cout << sol.validMountainArray(arr) << '\n';
  return 0;
}
